### PR TITLE
Add billing mode line

### PR DESCRIPTION
**Pull request by michael-jia-sage**
Add billing mode line

### CODE REVIEW

File: lib/dynamodb_framework/dynamodb_table_manager.rb [hunk 1]
```diff
@@ -257,6 +257,7 @@ def create(table_name, attributes, partition_key, range_key = nil, read_capacity
           :table_name => table_name,
           :attribute_definitions => attribute_definitions,
           :key_schema => key_schema,
+          :billing_mode => 'PAY_PER_REQUEST',
```
Comment (chrisbarber86) [inline lib/dynamodb_framework/dynamodb_table_manager.rb]: This needs to be an option passed into the method, it shouldn't default to `PAY_PER_REQUEST`. The default in the method signature should be `PROVISIONED`

Comment (chrisbarber86) [inline lib/dynamodb_framework/dynamodb_table_manager.rb]: This needs to be an option passed into the method, it shouldn't default to `PAY_PER_REQUEST`. The default in the method signature should be `PROVISIONED`

File: lib/dynamodb_framework/dynamodb_table_manager.rb [hunk 2]
```diff
@@ -257,6 +257,7 @@ def create(table_name, attributes, partition_key, range_key = nil, read_capacity
...
           :provisioned_throughput => {
```
Comment (chrisbarber86) [inline lib/dynamodb_framework/dynamodb_table_manager.rb]: when `billing_mode` is set to `PAY_PER_REQUEST`, the `provision_throughput` options shouldn't be specified

**Comment by chrisbarber86** [comment]
The `create_global_index` should be updated too: https://github.com/Sage/dynamodb_framework/blob/35b313ab9c5aa574bcdd17bc357712f96ed7e811/lib/dynamodb_framework/dynamodb_table_manager.rb#L341-L363

**Comment by chrisbarber86** [comment]
Also `create_table`: https://github.com/Sage/dynamodb_framework/blob/35b313ab9c5aa574bcdd17bc357712f96ed7e811/lib/dynamodb_framework/dynamodb_table_manager.rb#L278-L339

**Comment by taoliusage** [comment]
`billing_mode` needs to be included to match the input hash hierarchy as outlined in the AWS V2 doc:
https://docs.aws.amazon.com/sdkforruby/api/Aws/DynamoDB/Types/CreateTableInput.html
https://docs.aws.amazon.com/sdkforruby/api/Aws/DynamoDB/Types/UpdateTableInput.html

File: lib/dynamodb_framework/dynamodb_table_manager.rb [hunk 3]
```diff
@@ -364,11 +365,10 @@ def create_global_index(name, partition_key, range_key = nil, read_capacity = 20
           :key_schema => key_schema,
           :projection => {
               :projection_type => :ALL
-          },
-          :billing_mode => billing_mode
+          }
       }
 
-      unless billing_mode == ''
+      unless billing_mode == 'PAY_PER_REQUEST'
```
Comment (chrisbarber86) [inline lib/dynamodb_framework/dynamodb_table_manager.rb]: [Minor] This could just be `if billing_mode == 'PROVISIONED'`

Comment (taoliusage) [reply lib/dynamodb_framework/dynamodb_table_manager.rb]: Addressed.

**Pull request merged**